src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -1399,25 +1399,43 @@ bool NetworkOPsImp::checkLastClosedLedger (
 }
 
 void NetworkOPsImp::switchLastClosedLedger (
-    Ledger::pointer newLedger, bool duringConsensus)
+    Ledger::pointer newLCL, bool duringConsensus)
 {
-    // set the newledger as our last closed ledger -- this is abnormal code
+    // set the newLCL as our last closed ledger -- this is abnormal code
 
     auto msg = duringConsensus ? "JUMPdc" : "JUMP";
-    m_journal.error << msg << " last closed ledger to " << newLedger->getHash ();
+    m_journal.error << msg << " last closed ledger to " << newLCL->getHash ();
 
     clearNeedNetworkLedger ();
-    newLedger->setClosed ();
-    auto openLedger = std::make_shared<Ledger> (false, std::ref (*newLedger));
-    m_ledgerMaster.switchLedgers (newLedger, openLedger);
+    newLCL->setClosed ();
+    auto newOL = std::make_shared<Ledger> (false, std::ref (*newLCL));
+    // Caller must own master lock
+    {
+        // Apply tx in old open ledger to new
+        // open ledger. Then apply local tx.
+        auto const oldOL =
+            m_ledgerMaster.getCurrentLedger();
+        // VFALCO What do we use for the hash?
+        CanonicalTXSet retries({});
```

Comment:
I think it's fine.

If you don't set the salt, the entries are inserted in increasing order of `AccountID`, then sequence, then transaction ID.

If you do set the salt, same order except you're using a salted `AccountID`.

Now, I'm actually very skeptical about the way the salt is used, because it seems to me that the order _after_ salting is fairly strongly correlated to the order _before_ salting (e.g. one time in eight the first 3 bits of the salt are zero, in which case at least 7/8 of the comparisons are the same before and after).

But that's irrelevant for this application, where it's fine if the order is the same all the time.